### DISCUSSION TITLE
Extending Swell storefront editor widgets

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#521] [Help] [answered]
A client asks for the possibility to add new components for the Editor like Shopify but based on the Swell portal I found it [impossible](https://www.swell.is/help/storefronts-and-checkout-overview/themes/using-the-editor#accessing-the-editor) because the Editor depends on the themes which are no longer updated!

Any confirmation or solution?

**Comment by logeshswell** [comment]
Yes, you can add new components to the Swell storefront editor using one of the themes (Origin/Horizon).

But, as you described, we no longer support for new features or bug fixes for either themes.

**Reply by tarikrital** [reply]
Is there a reference for that?

**Reply by logeshswell** [reply]
Yes, we have the general instructions in README.md of either repo. Also, deeper understanding of the theme codebase is required to implement this feature.

**Reply by tarikrital** [reply]
Thank you, we currently added many features to extend the user account board using `swell-node` and I found [this](https://developers.swell.is/storefronts/origin#types).

>Yes, we have the general instructions in README.md of either repo.

The client planning to extend it, can you confirm the Swell dashboard editor will output the widgets as expected?

**Reply by logeshswell** [reply, marked as answer]
Yes, Swell storefront editor will output the custom widgets and a few merchants have already implemented this feature successfully for their storefront.

**Reply by tarikrital** [reply]
Thank you, @logeshswell.
